version: "3.9"

services:
  postgres:
    image: 'postgres'
    restart: always
    env_file:
      - ./.env
    volumes:
      - ./data/postgres-data/db:/var/lib/postgresql/data
      - ./data/postgres-data/backups:/backups
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

  bot:
    build: .
    command: python app.py
#    restart: on-failure
    volumes:
      - ./src/:/usr/src/
    env_file:
      - ./.env
    environment:
      POSTGRES_HOST: postgres
      REDIS_HOST: redis
      REDIS_PORT: 6379
    depends_on:
      redis:
        condition: service_started
      postgres:
        condition: service_healthy


  adminer:
    image: adminer
    restart: always
    ports:
      - 8080:8080
    depends_on:
      - postgres

  redis:
    image: redis:5-alpine
    restart: on-failure
    volumes:
    - ./data/redis-data:/data